test/models.spec.ts

```diff
@@ -695,6 +720,7 @@ describe("Unit | Filters", function () {
 
       // Assert
       expect(models.getFilterType(testData.basicFilter.toJSON())).toBe(models.FilterType.Basic);
+      expect(models.getFilterType(testData.basicFilterWithKeys.toJSON())).toBe(models.FilterType.Basic);
```

Comment:
please make sure this test passes. if yes, it is the most important in your changes.